### PR TITLE
Update with DQL v1 on SDK 4.5.0

### PR DESCRIPTION
**Pull request by magicwave**
Refactored original demo app using the new DQL query language feature.

In the original implementation, the app centered around "Tasks". This commit also renames the project, directory, and data model to use "ToDo"s. This is because the `DittoStore.execute()` function in Swift is `async`, and calling an async function in Swift uses the Swift `Task` object, the name of which collided with the original data model name.

### CODE REVIEW

**Comment by magicwave** [comment]
With this DQL refactor I also took a few liberties with the UX. I'm not sure choosing a name from the picker in the original implementation actually updated the subscription and (with DQL, the) store observer when selected. Or I may not have understood the intent. However, I made the two features more explicit by adding a menu to the plus button splitting the create-new-task and set-user features. The user experience is more intuitive as far as I understand the intent of those features, but I'm open to being corrected or to suggestions for improvement.

**Comment by magicwave** [comment]
Additionally, I'm using `QueryResultItem.jsonString()` to initialize `Codable` data model instances, more as an experiment to become familiar with DQL than necessarily a best practice. In previous work on this first DQL effort I implemented an initializer in the model that took a `[String: Any?]` argument, similar to the convenience initializer I've implemented in other demo apps taking a `DittoDocument` argument. 

I don't know that there's any particular benefit of one over the other, and I don't think there's much difference in the amount of code between the implementations. On the one hand, encoding the JSON string to data and then decoding into a model instance may involve a bit more processing (not sure); a possible advantage is type safety enforced by the `Codable` protocol over the explicit Swift casts from `Any?`. Open to thoughts on this too.

File: ToDo/EditScreen.swift [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+    func save() async {        
+        if let _id = _id { // updating existing ToDo
+            let query = "UPDATE tasks SET isCompleted = \(isCompleted)"
+            + ", userId = '\(userId)'"
+            + " WHERE _id=='\(_id)'"
+            
+            do {
+                try await dittoStore.execute(query: query)
```
Comment (danielhenrymantilla) [inline ToDo/EditScreen.swift:41]: So, this is more of a question for myself more than anything else, but I was wondering what the recommended style for these things would be: string interpolation or "aggressive" use of `queryArguments`?

```suggestion
            let query = "UPDATE tasks SET isCompleted = :completed"
            + ", userId = :userId"
            + " WHERE _id==:_id"
            
            do {
                try await dittoStore.execute(query: query, arguments: [
                    "isCompleted": isCompleted,
                    "userId": userId,
                    "_id": _id,
                ])
```

I guess this remark may become especially relevant the moment the values themselves may contain escaping charactes, _e.g._:

```swift
let example = "Daniel's example"
store.execute(query: "SELECT * FROM COLLECTION items WHERE sentence == '\(example)'")
```

Comment (magicwave) [reply ToDo/EditScreen.swift:41]: I went with what first came to mind, but I'm all about discovering best practices as I go. Great question. What is your opinion?
cc @skylerjokiel

Comment (okdistribute) [reply ToDo/EditScreen.swift:41]: Probably best to use query arguments to showcase the feature

**Review by okdistribute — CHANGES_REQUESTED** [review]
It would be ideal to have a diff for the Ditto Manager and Tasks model so you can see what is needed when moving from builder to DQL

**Comment by magicwave** [comment]
Created new PR with file original file names, except for renaming Task.swift to TaskModel.swift, and some enhancements.

**Pull request closed without merging**